From af5b353a34cc39fe44971da02f1fd16cb852fbbb Mon Sep 17 00:00:00 2001
From: gitlawr <lawrleegle@gmail.com>
Date: Fri, 6 Nov 2020 16:46:37 +0800
Subject: [PATCH 2/2] load image

---
 install.sh | 42 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/install.sh b/install.sh
index 0bc2376..1146a5a 100755
--- a/install.sh
+++ b/install.sh
@@ -40,6 +40,7 @@ cleanup2()
 
     losetup -d ${ISO_DEVICE} || losetup -d ${ISO_DEVICE%?} || true
     umount $DISTRO || true
+    sleep 10
 }
 
 cleanup()
@@ -163,6 +164,47 @@ do_copy()
         echo "Decompressing container images"
         zstd -d --rm "${root_path}/${offline_image_path}.zst" -o "${root_path}/${offline_image_path}" > /dev/null
     fi
+    echo "Loading images"
+    sleep 5
+    cd ${root_path}
+    mkdir lib bin sbin k3os dev proc etc sys
+    mount --bind /bin bin
+    mount --bind /sbin sbin
+    mount --bind /run/k3os/iso/k3os k3os
+    mount --bind /dev dev
+    mount --bind /proc proc
+    mount --bind /etc etc
+    mount -r --rbind /lib lib
+    mount -r --rbind /sys sys
+    chroot . /bin/bash
+
+    # invoke k3s to create data dir
+    k3s agent --flannel-backend=none &>/dev/null
+    # start containerd
+    /var/lib/rancher/k3s/data/*/bin/contaienrd \
+    -c /var/lib/rancher/k3s/agent/etc/containerd/config.toml \
+    -a /run/k3s/containerd/containerd.sock \
+    --state /run/k3s/containerd \
+    --root /var/lib/rancher/k3s/agent/containerd &
+
+    until ctr version>/dev/null
+    do
+      echo "waiting containerd to be ready"
+      sleep 1
+    done
+    # import images
+    ctr images import /var/lib/rancher/k3s/agent/images/*
+    rm /var/lib/rancher/k3s/agent/images/*
+    # stop containerd
+    pkill containerd
+    # exit chroot
+    exit
+
+    #cleanup
+    umount bin sbin k3os dev proc etc
+    mount --make-rslave {lib,sys}
+    umount -R {lib,sys}
+    rm -r lib bin sbin k3os dev proc etc sys
 }
 
 install_grub()
-- 
2.20.1 (Apple Git-117)

