From 1d2b265d3beb274cee758f1c391cab226fcef238 Mon Sep 17 00:00:00 2001
From: gitlawr <lawrleegle@gmail.com>
Date: Thu, 5 Nov 2020 16:11:24 +0800
Subject: [PATCH 2/3] debug

---
 install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install.sh b/install.sh
index 0bc2376..0993afc 100755
--- a/install.sh
+++ b/install.sh
@@ -363,5 +363,5 @@ else
     echo " * Installation completed"
     echo " * Rebooting system in 5 seconds"
     sleep 5
-    reboot -f
+    #reboot -f
 fi
-- 
2.20.1 (Apple Git-117)

