From b3107d295be305f04a1102396296e654566ea69a Mon Sep 17 00:00:00 2001
From: gitlawr <lawrleegle@gmail.com>
Date: Thu, 5 Nov 2020 17:50:14 +0800
Subject: [PATCH 3/3] debug

---
 install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install.sh b/install.sh
index 0993afc..bc575d0 100755
--- a/install.sh
+++ b/install.sh
@@ -343,7 +343,7 @@ fi
 validate_progs
 validate_device
 
-trap cleanup exit
+#trap cleanup exit
 
 get_iso
 setup_style
-- 
2.20.1 (Apple Git-117)

